src/containers/tutor-home-page/general-info-step/GeneralInfoStep.tsx

```diff
@@ -46,24 +43,70 @@ const GeneralInfoStep: FC<GeneralInfoStepProps> = ({ btnsBox }) => {
     handleNonInputValueChange,
     resetData
   } = useStepContext()
-  const countries = countriesMock
+  const [countries, setCountries] = useState<CountryType[]>([])
   const [cities, setCities] = useState<CityType[]>(
-    stepData !== stepDataInitialValues && stepData.country
-      ? stepData.country.cities
-      : []
+    stepData.country?.cities ?? []
   )
 
-  const handleCountryChange = (
+  const basePath = import.meta.env.VITE_API_BASE_PATH
+
+  useEffect(() => {
```

Comment:
we also should add the request to get cities in this useEffect (on component mount) 
like this: 
```
 if (stepData !== stepDataInitialValues && stepData.country) {
  const response = await axios.get<{
    label: string
    cities: CityType[]
  }>(`${basePath}/api/location/countries/${stepData.country.iso2}/cities`, {
    params: { countryName: stepData.country.label }
  })
  setCities(response.data.cities)
}
```

to be able to render cities in this scenario:
- the country were selected and the city is not, 
- then we move to another step (so general info step component gets unmounted)
- we go back and want to select a city (but right now we don't have any options in this case)